webtau-core/src/main/java/com/twosigma/webtau/reporter/TestStep.java

```diff
@@ -45,7 +45,7 @@
     public static <E> TestStep<E> create(E context,
                                          TokenizedMessage inProgressMessage,
                                          Supplier<TokenizedMessage> completionMessageSupplier,
-                                         Runnable action) {
+                                         Supplier action) {
```

Comment:
Presumably this supplier always supplies an R?  If so, couldn't we parameterise TestStep on it?  Not really fond of using generic classes with no generic params (not least because IntelliJ moans at you for it 😄 )

Reply:
I thought to add it, the problem is the actual type we only now at the `step.execute()` time and not the create time.

```
ThisIsTheType id = http.post(...) { 
    return id
}
```

so we can define it as a generic, but it won't do any good. Of course I may be missing some generic magic knowledge.